buildSrc/src/main/kotlin/OnlyAffectedTestTask.kt

```diff
@@ -70,6 +71,30 @@ open class OnlyAffectedTestTask : DefaultTask() {
         }
     }
 
+    /**
+     * Get the predicate to compute if the module should be included or not in the test
+     */
+    private fun computeModulesPredicate(allTests: Boolean, modulesPredicate: Boolean) = when {
+        allTests == true -> true
+        else -> modulesPredicate
+    }
+
+    /**
+     * Check if we have to run all tests, by looking at changes on libs versions
+     */
+    private fun hasToRunAllTests(): Boolean {
+        val isVersionsFileChanged =
+            "bash -c 'git diff --quiet origin/develop -- ${project.rootDir}/gradle/libs.versions.toml ; echo $?'".runCommandWithExitCode()
+        val isRootBuildFileChanged =
+            "bash -c 'git diff --quiet origin/develop -- ${project.rootDir}/build.gradle.kts ; echo $?'".runCommandWithExitCode()
+
+        return (isVersionsFileChanged != 0 || isRootBuildFileChanged != 0).also {
+            if (it) {
+                println("\uD83D\uDD27 Running all tests because there are changes at the root level")
+            }
+        }
```

Comment:
`settings.gradle.kts` too?

Maybe

```kotlin
        val globalBuildSettingsFiles = listOf(
            "${project.rootDir}/gradle/libs.version.toml",
            "${project.rootDir}/build.gradle.kts",
            "${project.rootDir}/settings.gradle.kts",
        )
        val anySettingsFileChanged = globalBuildSettingsFile.any { it != 0 }
        if (anySettingsFileChanged) {
            println("\uD83D\uDD27 Running all tests because there are changes at the root level")
        }
        return anySettingsFileChanged
```